llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -4561,6 +4577,10 @@ bool SROA::presplitLoadsAndStores(AllocaInst &AI, AllocaSlices &AS) {
         PStore->copyMetadata(*SI, {LLVMContext::MD_mem_parallel_loop_access,
                                    LLVMContext::MD_access_group,
                                    LLVMContext::MD_DIAssignID});
+
+        if (AATags)
+          PStore->setAAMetadata(
```

Comment:
Is this part covered in a testcase ?

Reply:
Should be now, I added a number of additional tests that should cover all cases here

Reply:
thx !